pdns/lua-record.cc

```diff
@@ -1103,6 +1115,36 @@ static void setupLuaRecords(LuaContext& lua)
       return result;
     });
 
+  lua.writeFunction("dblookup", [](const string& record, const string& type) {
+    DNSName rec;
+    QType qtype;
+    vector<string> ret;
+    try {
+      rec = DNSName(record);
+      qtype = type;
```

Comment:
`QType &operator=(const string &)` does not throw an exception on error, but assigns zero.

Reply:
good catch, will handle

Reply:
```suggestion
      qtype = type;
      if (qtype.getCode() == 0) {
        throw std::invalid_argument("unknown type");
      }
```

plus leave `type` in the next log line?

Reply:
yes